src/Grid/grid.jl

```diff
@@ -1320,8 +1320,22 @@ end
     return x
 end
 
+@inline function getcoordinates!(x::Vector{Vec{dim,T}}, grid::Ferrite.AbstractGrid, (cellid, lfaceid)::FaceIndex) where {dim,T} 
+    cell = getcells(grid, cellid)
+    _getfacecoordinates!(x, grid, cell, lfaceid) #function barier for `cell`
+end
+
+@inline function _getfacecoordinates!(x::Vector{Vec{dim,T}}, grid::Ferrite.AbstractGrid, cell::Ferrite.AbstractCell, lfaceid::Int) where {dim,T}
+    ip = default_interpolation(typeof(cell))
+    faceindices = Ferrite.facedof_indices(ip)[lfaceid]
```

Comment:
`faces` only export the vertices/corner-nodes of the cell (even for QuadraticHexahedrons etc). So we would need to update `faces(cells) `in that case...

Reply:
I think we should not touch these definitions (i.e. `faces`, because they already give the expected description of the corresponding topological entity and this is what we actually need in most cases.

Reply:
I should also note that AFAIK there is no hard guarantee that the returned tumple of `facedof_indices` must match the exact ordering of the face nodes.

Reply:
Would be nice to have a good interface for getting the dofs on a face, this is also used in this function: 
https://github.com/Ferrite-FEM/Ferrite.jl/blob/c6c185cc3364892777f5a9b72a8f30a04db2c7bb/src/Dofs/ConstraintHandler.jl#L981
(when I needed it once I would have wanted such an interface)

Reply:
@termi-official  I dont know what makes most sense: `faces(QuadraticHexhedron)` returning only the 4 corner nodes or all 9 nodes on the face. 

But I think the cell-ordering has to follow `facedof_indices` right?, otherwise the dof-distrubtion would be wrong I think.

Reply:
Note that the purpose of `facedof_indices` is different from `facedof_interior_indices`. We only need the latter during serial dof distribution. The latter is just a query of "give me every dof in the face", which is a cherry pick of what I need in the distributed assembly PR. I think it makes sense to somehow enforce that the order of the dofs is aligned with the subentity order on the corresponding cell, but I need to check exactly which invariants I will require for the dof synchronization in the new API. Does this clear up the confusion?

Reply:
Not sure if I understand the first one correctly. `faces(QuadraticHexhedron)` should return the 6 faces with each face described by its corners. This is everything that we need to derive the relevant topological information as well as orientation information. The higher order nodes give you "just" the extra geometric information about how the element deforms on this specific face. But yea, we should implement some additional API to query higher order information for the geometry, even if it is just by enforcing a strict correspondence between the element ad the default interpolation (which I think Kim did vote against previously).

Reply:
Just a reminder (not sure if it's needed here), `Ferrite.facedof_indices` is not defined for discontinuous interpolations. In case it's needed, `Ferrite.dirichlet_boundarydof_indices` was added in #729 . It might sound out of place here tho.